From 9139b539fe60397972f949fb9768a96aac321ea7 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 11 Apr 2022 13:12:40 +0300
Subject: [PATCH] [SW-66329] pif_arc: add specific implementation for dbg mode check

hl_is_debug_mode() function deciding on the debug mode by reading the
strap register. Since pif arc is not allowed to read this register,
a specific implementation is required for the pif arc.
For the moment the implementation is empty and not debug mode is
supported by the pif arc. in case some debug mode will required in
the future, some alternative method should be implmented to signal
the pif are about the required mode.

Change-Id: Id5dab0630d385403a1ad6de90ea341c0fbdc8c3b
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/app/pcie_pif/pcie_pif.c b/app/pcie_pif/pcie_pif.c
index 89e34d4..63f33f2 100644
--- a/app/pcie_pif/pcie_pif.c
+++ b/app/pcie_pif/pcie_pif.c
@@ -23,6 +23,8 @@
 	},
 };
 
+bool hl_dev_spec_debug_mode(enum hl_debug_hw_func func) { return false; }
+
 static void device_init(void)
 {
 	device_init_global_conf();
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 1098a55..04437c6 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -148,7 +148,7 @@
 	image_entry();
 }
 
-bool hl_is_debug_mode(enum hl_debug_hw_func func)
+__weak bool hl_dev_spec_debug_mode(enum hl_debug_hw_func func)
 {
 	u32 boot_strap = regmap_readl(mmD0_PSOC_BOOT_CONF_BOOT_STRAP_PINS_L);
 	u32 strap_mask;
@@ -172,6 +172,11 @@
 	return boot_strap & strap_mask;
 }
 
+bool hl_is_debug_mode(enum hl_debug_hw_func func)
+{
+	return hl_dev_spec_debug_mode(func);
+}
+
 u16 hl_get_boot_info_ver(void)
 {
 	return HL_BOOT_INFO_VERSION;
diff --git a/include/preboot.h b/include/preboot.h
index 623f7ad..a10e397 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -142,6 +142,7 @@
 void hl_ts_shutdown(void);
 void hl_plat_init(void);
 bool hl_is_debug_mode(enum hl_debug_hw_func func);
+bool hl_dev_spec_debug_mode(enum hl_debug_hw_func func);
 bool hl_is_jtag_enabled(void);
 void hl_enter_recovery(void);
 void hl_dev_reset(void);
